Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,355 @@
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestBalanceEver = decimal.MinValue;
+            var PersonWithHighestBalance = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                HighestBalanceEver = FindHighestBalance(peopleAndBalances, HighestBalanceEver, PersonWithHighestBalance, i);
+
+            }
+
+            var answer = $"{PersonWithHighestBalance.ToString()} had the most money ever. ¤{HighestBalanceEver}.";
+
+            return answer;
+        }
+
+        private static decimal FindHighestBalance(string[] peopleAndBalances, decimal HighestBalanceEver, StringBuilder PersonWithHighestBalance, int i)
+        {
+            var currentPersonData = peopleAndBalances[i].Split(',');
+            var highestAmountOfPerson = FindHighestBalanceOfPerson(currentPersonData);
+
+            if (highestAmountOfPerson > HighestBalanceEver)
+            {
+                HighestBalanceEver = highestAmountOfPerson;
+                var currentOne = PersonWithHighestBalance.ToString();
+
+                if (string.IsNullOrEmpty(currentOne))
+                {
+                    PersonWithHighestBalance.Append(currentPersonData[0]);
+                }
+                else
+                {
+                    PersonWithHighestBalance.Clear();
+                    PersonWithHighestBalance.Append(currentPersonData[0]);
+                }
+
+
+            }
+            else if (highestAmountOfPerson == HighestBalanceEver)
+            {
+                if (i == peopleAndBalances.Length - 1)
+                {
+                    PersonWithHighestBalance.Append(" and ");
+                }
+                else
+                {
+                    PersonWithHighestBalance.Append(", ");
+                }
+
+                PersonWithHighestBalance.Append(currentPersonData[0]);
+            }
+
+            return HighestBalanceEver; 
+        }
+
+        public static decimal FindHighestBalanceOfPerson(string[] currentPersonData)
+        {
+            var highest = decimal.MinValue; 
+            var allAmounts = currentPersonData[1..]; 
+            foreach (var amountString in allAmounts)
+            {
+                var isValid = decimal.TryParse(amountString, out decimal amount); 
+                if (amount > highest)
+                    
+                {
+                    highest = amount; 
+                }
+            }
+
+            return highest; 
+        }
+
+        private static decimal CheckIfMorePeopleHaveTheSameAmount(decimal highestBalanceEver, StringBuilder personWithHighestBalance, string[] currentPersonData, decimal highestCurrentPersonBalance)
+        {
+            if (highestCurrentPersonBalance > highestBalanceEver)
+            {
+                highestBalanceEver = highestCurrentPersonBalance;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'highestBalanceEver'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)

Reply:
How to solve this. This seems legal to me